appinventor/components-ios/src/facemesh.swift

```diff
@@ -0,0 +1,760 @@
+  }
+  
+  @objc public func setModel(_ path: String) {
+
+      if path.hasSuffix(MODEL_PATH_SUFFIX) {
+          // Set the model path to the provided path
+        _modelPath = path
+      } else {
+        _form?.dispatchErrorOccurredEvent(self, "Model", ErrorMessage.ERROR_CANNOT_FIND_FILE, "\(FaceExtension.ERROR_INVALID_MODEL_FILE): Invalid model file format. Files must be of format \(MODEL_PATH_SUFFIX)")
+      }
+  }
+  
+//    open func ClassifierReady(){}
+//    open func GotClassification(_ result: AnyObject){}
+//    open func Error(_ errorCode: Int32){}
+
+  
+  private func configureWebView(_ webview: WKWebView) {
+      _webview = webview
+      let config = webview.configuration
+
+      if #available(iOS 16.4, *) {
+          webview.isInspectable = true
+      }
+
+      config.preferences.javaScriptEnabled = true
+      config.allowsInlineMediaPlayback = true
+      config.mediaTypesRequiringUserActionForPlayback = []
+      
+      if self is FaceExtension {
+          print("FaceExtension detected")
+          _webview!.configuration.userContentController.add(self, name: "FaceExtension")
+      } else {
+          // Implement checks for other AI components if needed
+          print("Other AI component detected")
+      }
+  }
+
+//    private func parseLabels(_ labels: String) throws -> [String] {
+//      var result = [String]()
+//      let data = Data(labels.utf8)
+//      do {
+//        if let arr = try JSONSerialization.jsonObject(with: data, options: []) as? [Any] {
+//          for item in arr {
+//            result.append(labels)
+//          }
+//        } else {
+//          throw YailRuntimeError("Got unparsable array from Javascript", "RuntimeError")
+//        }
+//      } catch {
+//        throw YailRuntimeError("Got unparsable array from Javascript", "RuntimeError")
+//      }
+//      return result
+//    }
+    
+    //      internal func assertWebView(_ method: String, _ frontFacing: Bool = true) throws {
+    //        guard let _webview = _webview else {
+    //          throw AIError.webviewerNotSet
+    //        }
+    //      }
+    
+  @objc open var WebViewer: WebViewer {
+    get {
+      return _webviewer!
+    }
+    set {
+      newValue.aiSchemeHandler = self
+      configureWebView(newValue.view as! WKWebView)
+      print("configureWebView called")
+      if self is FaceExtension{
+        assetPath = "index"
+      }else{
+        // implement checks for other AI components
+      }
+      if let url = Bundle(for: FaceExtension.self).url(forResource: "index", withExtension: "html") {
+        let readAccessURL = Bundle(for: FaceExtension.self).bundleURL
+        let request = URLRequest(url: URL(string: "appinventor://localhost/\(assetPath ?? "").html")!)
+        //        let request = URLRequest(url: url)
+        print(request)
+        _webview?.load(request)
+        print("Request loaded")
+      }else {
+        print("request not loaded")
+      }
+            showImagePicker()
```

Comment:
can you align the number of shifts? in mac all you need to do is `command + a` and then `control + i`